### PR TITLE
[bitnami/elasticsearch] Protocol selection 

### PR DESCRIPTION
**Pull request by mFranz82**
**Description of the change**

When working with **Istio** the port naming must be observed in order to make **port selection** work. 

If one needs to access the elastic search services within a service mesh and **security is enabled** the port name needs to reflect the protocol. **With Kubernetes > 1.18 the field appProtocol can be used for this purpose.**

**Benefits**

Istio in conjunction with sidecar configurations works as aspected.

**Possible drawbacks**

If you already have a cluster running with security enabled the **Service** and corresponding **Pod port names** will change from **http** to **https**.

**Applicable issues**

No 

**Additional information**

https://istio.io/latest/docs/ops/configuration/traffic-management/protocol-selection/

**Checklist**
- [X] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [X] Variables are documented in the values.yaml and added to the README.md using (readme-generator-for-helm)[https://github.com/bitnami-labs/readme-generator-for-helm]
- [X] Title of the PR starts with chart name (e.g. [bitnami/<name_of_the_chart>])
- [X] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/master/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

File: bitnami/elasticsearch/templates/coordinating-statefulset.yaml [hunk 1]
```diff
@@ -161,7 +161,7 @@ spec:
             {{- end }}
           {{- end }}
           ports:
-            - name: http
+            - name: {{ if .Values.security.enabled }}https{{ else }}http{{ end }}
```
Comment (fmulero) [inline bitnami/elasticsearch/templates/coordinating-statefulset.yaml:164]: It seems the same change is needed in several places, Could you move this logic to the `_helpers.tpl` file?

**Review by fmulero — COMMENTED** [review]
This change could help a lot of users . Thank a lot @mFranz82.
I left a litle comment, just only to keep the code more maintainable

**Comment by mFranz82** [comment]
@fmulero You are absolutely right 👍 
Could you please have a look again?

**Comment by fmulero** [comment]
Great!. LGTM

**Pull request merged**